crates/astria-sequencer/src/service/consensus.rs

```diff
@@ -80,44 +83,40 @@ impl Consensus {
                     .context("failed initializing chain")?,
             ),
             ConsensusRequest::PrepareProposal(prepare_proposal) => {
-                ConsensusResponse::PrepareProposal(response::PrepareProposal {
-                    txs: prepare_proposal.txs,
-                })
-            }
-            ConsensusRequest::ProcessProposal(_process_proposal) => {
-                // TODO: handle this
-                ConsensusResponse::ProcessProposal(response::ProcessProposal::Accept)
+                ConsensusResponse::PrepareProposal(
+                    Self::prepare_proposal(prepare_proposal)
+                        .context("failed to prepare proposal")?,
+                )
             }
-            ConsensusRequest::BeginBlock(begin_block) => {
-                ConsensusResponse::BeginBlock(
-                    self.begin_block(begin_block)
-                        .await
-                        // .context() cannot be used here because Box<dyn Error> does not implement
-                        // Error: https://github.com/dtolnay/anyhow/issues/66
-                        .map_err(|e| anyhow!(e))
-                        .context("failed to begin block")?,
+            ConsensusRequest::ProcessProposal(process_proposal) => {
+                ConsensusResponse::ProcessProposal(
+                    match Self::process_proposal(&process_proposal) {
+                        Ok(()) => response::ProcessProposal::Accept,
+                        Err(e) => {
+                            warn!(error = ?e, "rejecting proposal");
+                            response::ProcessProposal::Reject
```

Comment:
Meh, sucks that there is no way to communicate why a proposal was rejected.

Reply:
This is by design of ABCI2 (https://github.com/cometbft/cometbft/blob/main/spec/abci/abci++_methods.md#processproposal). 

This is part of the decentralized consensus process, if running an honest validator there should be agreement and your local node will have context on why declined.